examples/rbfe_edge_list.py

```diff
@@ -0,0 +1,129 @@
+import argparse
+import csv
+import pickle
+
+import numpy as np
+from rdkit import Chem
+
+Chem.SetDefaultPickleProperties(Chem.PropertyPickleOptions.AllProps)
+
+from timemachine.fe import atom_mapping
+from timemachine.fe.rbfe import HostConfig, estimate_relative_free_energy
+from timemachine.ff import Forcefield
+from timemachine.md import builders
+from timemachine.parallel.client import CUDAPoolClient
+
+
+def get_mol_by_name(mols, name):
+    for m in mols:
+        if m.GetProp("_Name") == name:
+            return m
+
+    assert 0, "Mol not found"
+
+
+def run_pair(mol_a, mol_b, core, forcefield_path, protein_path, n_frames):
+
+    box_width = 4.0
+    solvent_sys, solvent_conf, solvent_box, solvent_top = builders.build_water_system(box_width)
+    solvent_box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+    solvent_host_config = HostConfig(solvent_sys, solvent_conf, solvent_box)
+
+    forcefield = Forcefield.load_from_file(forcefield_path)
+
+    solvent_res = estimate_relative_free_energy(
+        mol_a, mol_b, core, forcefield, solvent_host_config, n_frames=n_frames, prefix="solvent"
+    )
+
+    complex_sys, complex_conf, _, _, complex_box, complex_top = builders.build_protein_system(protein_path)
+    complex_box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+    complex_host_config = HostConfig(complex_sys, complex_conf, complex_box)
+
+    complex_res = estimate_relative_free_energy(
+        mol_a, mol_b, core, forcefield, complex_host_config, n_frames=n_frames, prefix="complex"
+    )
+
+    return solvent_res, solvent_top, complex_res, complex_top
+
+
+def read_from_args():
+
+    parser = argparse.ArgumentParser(
+        description="Estimate relative free energy difference between complex and solvent given two ligands mol_a and mol_b."
+    )
+    parser.add_argument(
+        "--n_frames", type=int, help="number of frames to use for the free energy estimate", required=True
+    )
+    parser.add_argument("--ligands", type=str, help="SDF file containing the ligands of interest", required=True)
+    parser.add_argument("--results_csv", type=str, help="Results containing all the csv", required=True)
+    parser.add_argument("--forcefield", type=str, help="location of the ligand forcefield", required=True)
+    parser.add_argument("--protein", type=str, help="PDB of the protein complex", required=True)
+    parser.add_argument("--n_gpus", type=int, help="number of gpus", required=True)
+
+    args = parser.parse_args()
+
+    mols = [mol for mol in Chem.SDMolSupplier(str(args.ligands), removeHs=False)]
+
+    futures = []
+    metadata = []
+
+    cpc = CUDAPoolClient(args.n_gpus)
+    cpc.verify()
+
+    with open(args.results_csv) as csvfile:
+        reader = csv.reader(csvfile, delimiter=",")
+        next(reader)
+        rows = [row for row in reader]
+        for row in rows:
+            mol_a_name, mol_b_name, exp_ddg, fep_ddg, fep_ddg_err, ccc_ddg, ccc_ddg_err = row
```

Comment:
Should the format be more flexible (allow exp_ddg, ... to be empty strings)?

Reply:
probably - this file only works on the pre-generated edge list right now, but we can change it later.